redfish-core/lib/cable.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+#include <boost/container/flat_map.hpp>
+#include <utils/json_utils.hpp>
+
+namespace redfish
+{
+using ManagedPropertyType =
+    boost::container::flat_map<std::string, std::variant<std::string, double>>;
+
+/**
+ * @brief Get Cable properties
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       resp        A map to hold Cable properties
+ * @param[in]       cableId     Cable Id
+ */
+inline void
+    getCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       ManagedPropertyType& resp, const std::string& cableId)
+{
+    double* cableLength = nullptr;
+    std::string* cableTypeDescription = nullptr;
+
+    for (auto& propertyMap : resp)
+    {
+        if (propertyMap.first == "CableTypeDescription")
+        {
+            cableTypeDescription =
+                std::get_if<std::string>(&propertyMap.second);
+        }
+        else if (propertyMap.first == "Length")
+        {
+            cableLength = std::get_if<double>(&propertyMap.second);
+        }
+    }
+
+    if (cableTypeDescription == nullptr || cableLength == nullptr)
+    {
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    std::string dataId = "/redfish/v1/Cables/" + cableId;
+    asyncResp->res.jsonValue["@odata.type"] = "#Cable.v1_0_0.Cable";
+    asyncResp->res.jsonValue["@odata.id"] = dataId;
+    asyncResp->res.jsonValue["Id"] = cableId;
+    asyncResp->res.jsonValue["Name"] = "Cable Id";
+    asyncResp->res.jsonValue["LengthMeters"] = *cableLength;
```

Comment:
If these are not mandatory properties to be filled in schema then we can fill them directly at line 27 and 32. This way we can get rid of check at line 36 and variable at line 20 and 21.

Reply:
I'm not sure whether they are mandatory or not. If they are mandatory then we need to have that check at line#36. Also, we need to have those variables at line#20 and 21 to get the corresponding properties if the properties are mandatory. Otherwise we need to make the change to line #27 and 32 to to make those as variables and then check if either one or both a nullptr. 

IMO, it is good to leave it as is since it is more readable and are at one place.

@gtmills, @bradbishop : Are the description and cable length are mandatory? The default description is empty string and cable length is NaN.

Reply:
'"readonly" : true' in the schema means mandatory - correct?

Reply:
made the change taking into account whether these properties are mandatory or not...